src/renderer/components/+catalog/catalog-entity-detail-items/catalog-entity-detail-item-injection-token.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectionToken } from "@ogre-tools/injectable";
+import type { LensRendererExtension } from "../../../../extensions/lens-renderer-extension";
+import type { CatalogEntity } from "../../../../common/catalog";
+import type { CatalogEntityDetailItemComponentProps } from "./extension-registration";
+
+export interface CatalogEntityDetailItem<T extends CatalogEntity> {
+  apiVersions: string[];
+  kind: string;
+
+  components: {
+    Details: React.ComponentType<CatalogEntityDetailItemComponentProps<T>>;
+  };
+
+  orderNumber: number;
+
+  extension?: LensRendererExtension;
```

Comment:
Why is this optional?

Reply:
It's optional because there is no need to limit `catalog-entity-detail-item` for extensions. Same interface is used for detail-items that live in OpenLens. Notice that this relates to other comment.